### PR TITLE
E2E Tests in CI

### LINKED ISSUE #8433: Run GUI2 E2E tests (they do not include backend) as job on CI
More or less, as in the title. They should be run in the "test" step only.

Scope includes fixing them if they do not pass.

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

Fixes #8433 

* Adds E2E test to `test` script in gui2 (without server)
* Add options to `run` script to specify what test should be run: `unit` (default), `e2e`, or `ci` (which runs both unit and e2e without watching/spawning report server).
* The CI test step now checks e2e tests.

### Important Notes

~~One of e2e tests was disabled because it caught the regression on develop: #8476 ~~

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- ~~[ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.`~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - ~~[ ] Unit tests have been written where possible.~~
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: build/build/src/project/gui2.rs [hunk 1]
```diff
@@ -53,26 +53,36 @@ pub fn script(repo_root: impl AsRef<Path>, script: Scripts) -> Result<NpmCommand
...
+/// Run E2E tests.
+pub fn e2e_tests(repo_root: impl AsRef<Path>) -> BoxFuture<'static, Result> {
+    run_in_repo_root(Scripts::TestE2e, repo_root)
+}
+
+pub fn watch(repo_root: impl AsRef<Path>) -> BoxFuture<'static, Result> {
+    run_in_repo_root(Scripts::Dev, repo_root)
+}
+
+fn run_in_repo_root(script: Scripts, repo_root: impl AsRef<Path>) -> BoxFuture<'static, Result> {
```
Comment (Frizi) [inline build/build/src/project/gui2.rs]: How about renaming this to `install_and_run_script`?

File: build/cli/src/arg/gui2.rs [hunk 2]
```diff
@@ -15,16 +16,32 @@ source_args_hlp!(Gui2, "gui2", BuildInput);
 #[derive(Args, Clone, Copy, Debug, PartialEq)]
 pub struct BuildInput {}
 
+#[derive(ArgEnum, Clone, Copy, Debug, PartialEq)]
+pub enum TestType {
+    /// Run unit tests in watch mode
+    Unit,
+    /// Run E2E tests
+    E2E,
+    /// Run all tests without watching nor HTML reports.
```
Comment (Frizi) [inline build/cli/src/arg/gui2.rs]: Producing HTML reports for failing tests and adding them to the CI artifacts could maybe be useful. Something to check if it's worth the effort.

**Review by Frizi — APPROVED** [review]

File: build/cli/src/arg/gui2.rs (hunk 2 above)
Comment (farmaazon) [reply build/cli/src/arg/gui2.rs]: Let's do it in a separate PR

**Comment by farmaazon** [comment]
The visualization regression was fixed, so I enabled the test.

**Review by mwu-tow — APPROVED** [review]
Not blockers but for consideration:
* I don't like `ci` argument, as it does not describe *what* it is doing. I'd rather name it `all`.
* Shouldn't the CI-style test be the default? It might be counterintuitive that CI says that "GUI2 tests" failed, while `./run gui2 test` succeeds.
* In general, the `test` commands so far never created any watches — perhaps a better command would be `watch test` since we have `watch` command. (and then "test" would be what CI does, and "watch test" would be more self-descriptive)

**Pull request merged**